### ISSUE TITLE
epinio push --skip-ssl-verification crash if epinio is SSL-enabled

### ISSUE THREAD

**Issue by gitricko** [epinio/epinio#1603]
**Describe the bug**

If your epinio deployment is configure with TLS from let's encrypt and when you do `epinio push --skip-ssl-verification --name APPNAME`, you will get below error. Removing `--skip-ssl-verification` will make epinio push working again.

Question why ignore certificate is causing this issue.

```
....
⚠️  Hit Enter to continue or Ctrl+C to abort (deployment will continue automatically in 5 seconds)


Create the application resource ...

Application exists, updating ...

Collecting the application sources ...

Uploading application code ...

Staging application with code...

🕞  Running staging

🚢  Streaming application logs
Namespace: workspace
Application: expense-mgt-dev
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x5bcae3f]

goroutine 40 [running]:
github.com/epinio/epinio/pkg/api/core/v1/client.(*Client).AppLogs(0xc000114a80, {0xc000bf0620, 0x9}, {0x7ff7bfeff62b, 0xf}, {0xc000a1ca50, 0x10}, 0x1, 0xc0001356f8)
        github.com/epinio/epinio/pkg/api/core/v1/client/apps.go:395 +0x61f
github.com/epinio/epinio/internal/cli/usercmd.(*EpinioClient).AppLogs(0xc000bd65c0, {0x7ff7bfeff62b, 0xf}, {0xc000a1ca50, 0x10}, 0x0?)
        github.com/epinio/epinio/internal/cli/usercmd/app.go:390 +0x69f
github.com/epinio/epinio/internal/cli/usercmd.(*EpinioClient).stageLogs.func1()
        github.com/epinio/epinio/internal/cli/usercmd/push.go:256 +0x4a
created by github.com/epinio/epinio/internal/cli/usercmd.(*EpinioClient).stageLogs
        github.com/epinio/epinio/internal/cli/usercmd/push.go:255 +0xef

```

Hint: Make sure to search through the existing issues, as they often contain important information on recent changes.

**To Reproduce**
Steps to reproduce the behavior, e.g.:

see above

**Expected behavior**
Even if LetsEncrypt certificate is configure correctly in epinio deployment, ignore ssl should cause epinio push to fail.

**Logs**
If applicable, add logs to help explain your problem.

You can use attachments to add a screenshot of your clusters state, e.g. from k9s.

If you paste long logs, you could also add them into a collapsed block:

&lt;details&gt;
  &lt;summary&gt;Click to expand&lt;summary&gt;

  \```
  pasted log
  \```
&lt;details&gt;

You can increase Epinio's logging by exporting this variable:

```
export TRACE_LEVEL=255
```

To follow the server logs you can use:

`kubectl logs -n epinio -l app.kubernetes.io/name=epinio-server -c epinio-server -f`

**Cluster (please complete the following information):**
 - Provider: k3d
 - Kubernetes Version: [e.g. 1.23]

**Desktop/CLI (please complete the following information):**
 - OS: happen in both linux and osx
 - Epinio Version: 1.0.0

**Comment by juadk** [comment]
Thanks for the report! Do not expect a feedback this week, it's a special week at SUSE (Hackweek). Talk to you next week.

**Comment by andreas-kupries** [comment]
This could be

```
>  391:         websocketURL := fmt.Sprintf("%s%s/%s?%s", c.WsURL, api.WsRoot, endpoint, queryParams.Encode())
>  392:         webSocketConn, resp, err := websocket.DefaultDialer.Dial(websocketURL, http.Header{})
>  393:         if err != nil {
>  394:                 // Report detailed error found in the server response
>  395: --->            if resp.StatusCode != http.StatusOK {
>  396:                         defer resp.Body.Close()
>  397:                         bodyBytes, errBody := ioutil.ReadAll(resp.Body)
```

Assuming the code did not change since the report .... Not according to 
https://github.com/epinio/epinio/blame/main/pkg/api/core/v1/client/apps.go#L395
which was https://github.com/epinio/epinio/commit/4792df34f8c8e336cddcf96befa86a8825709ba6

So this could be some kind of error response from the dialer (`err != nil`) where the `resp` result is not set, i.e. `nil` ?!

I will create a PR which checks for `resp == nil` in that line. Then we can see if that was indeed the problem, and also see what the original error was.

@gitricko Are you willing to tell me what TZ you are in ?

**Comment by andreas-kupries** [comment]
As a side note, the crash does not happen with a local install not using LE. An LE-induced  error response seems to be required for this case.

**Comment by andreas-kupries** [comment]
And this PR, #1621, also passes the CI. See https://github.com/epinio/epinio/runs/7197290557?check_suite_focus=true#step:11:247

Conclusions:
  - I believe now that core of reported issue is a problem in the setup of the LE stuff.
  - This problem induced the crash via `err != nil, resp == nil` from the dialer.
  - The crash incidentally hides the core issue as the actual websocket error is not printed.

The PR #1620 likely fixes the crash. @gitricko, please test a binary from that branch in your environment.
This should also tell us the currently hidden error message.

**Comment by gitricko** [comment]
> And this PR, #1621, also passes the CI. See https://github.com/epinio/epinio/runs/7197290557?check_suite_focus=true#step:11:247
> 
> Conclusions:
> 
>     * I believe now that the core of reported issue is a problem in the setup of LE in the reporter's environment.
> 
>     * This problem induced the crash via `err != nil, resp == nil` from the dialer.
> 
>     * The crash incidentally hides the core issue as the actual websocket error is not printed.
> 
> 
> The PR #1620 likely fixes the crash. @gitricko, please test a binary from that branch in your environment. This should also tell us the currently hidden error message.

Hi @andreas-kupries,

if somebody or you can educate me how to download the binary from this branch, i can help to verify the fix.
Thanks

**Comment by andreas-kupries** [comment]
> how to download the binary from this branch

@gitricko 

Ah, I had hoped that you would be in a position to checkout the branch and compile the binary ...
(Needs a go installation, we have 1.18)

While I can build the binary the problem then becomes how to get it to you.
While I have a web server where I can place it for download I do not wish to have the url here up in public.
To get the url to you in a more private way ...
  - Are you somewhere on the SUSE slack ?
  - Maybe an email address ?

I am reachable at andreas.kupries@suse.com if you do not wish to provide your address here in public ?

**Comment by gitricko** [comment]
@andreas-kupries i reach out to you in suse/epinio/rancher slack then

**Comment by andreas-kupries** [comment]
I saw it, and have responded.

**Comment by andreas-kupries** [comment]
Considered as fixed by #1620.
